src/pub.c

```diff
@@ -279,10 +279,9 @@ natsConnection_PublishString(natsConnection *nc, const char *subj,
 natsStatus
 natsConnection_PublishMsg(natsConnection *nc, natsMsg *msg)
 {
-    const char *reply = (msg != NULL ? msg->reply : NULL);
```

Comment:
We need to rewrite this function and if `msg == NULL` return an invalid arg. The check of `nc` is done in the lower function, but could easily be added into this one.

As for removing the reply subject, of course we need it, but yes, the called function will take it from the provided message structure if not provided as a parameter, so I guess this change is ok.

Reply:
No, I meant what the contributor did, that is, check for `msg == NULL` as an invalid arg and return if that is the case. He did add the check for `nc == NULL` (although that is done in the internal function so that was not entirely necessary).